ecommerce/extensions/dashboard/orders/views.py

```diff
@@ -62,20 +63,30 @@ class OrderDetailView(CoreOrderDetailView):
     line_actions = ('change_line_statuses', 'create_shipping_event', 'create_payment_event', 'create_refund')
 
     def create_refund(self, request, order, lines, _quantities):  # pylint: disable=unused-argument
-        refund = Refund.create_with_lines(order, lines)
-
-        if refund:
-            data = {
-                'link_start': '<a href="{}" target="_blank">'.format(
-                    reverse('dashboard:refunds-detail', kwargs={'pk': refund.pk})),
-                'link_end': '</a>',
-                'refund_id': refund.pk
-            }
-            message = _('{link_start}Refund #{refund_id}{link_end} created! '
-                        'Click {link_start}here{link_end} to view it.').format(**data)
-            messages.success(request, mark_safe(message))
-        else:
-            message = _('A refund cannot be created for these lines. They may have already been refunded.')
+        if self._is_order_from_mobile(order):
+            message = _("Mobile payment refunds aren't allowed in ecommerce.")
             messages.error(request, message)
 
+        else:
+            refund = Refund.create_with_lines(order, lines)
+
+            if refund:
+                data = {
+                    'link_start': '<a href="{}" target="_blank">'.format(
+                        reverse('dashboard:refunds-detail', kwargs={'pk': refund.pk})),
+                    'link_end': '</a>',
+                    'refund_id': refund.pk
+                }
+                message = _('{link_start}Refund #{refund_id}{link_end} created! '
+                            'Click {link_start}here{link_end} to view it.').format(**data)
+                messages.success(request, mark_safe(message))
+            else:
+                message = _('A refund cannot be created for these lines. They may have already been refunded.')
+                messages.error(request, message)
+
         return self.reload_page()
+
+    def _is_order_from_mobile(self, order):
+        processor_responses = order.basket.paymentprocessorresponse_set.all()
+        return any(response.processor_name in MOBILE_PAYMENT_PROCESSORS
```

Comment:
I suspect MOBILE_PAYMENT_PROCESSORS is fairly small, but I think iterating over the `processor_responses` variable will make 1 db call for each item (unless it had been prefetched ahead of time). You might consider doing  something like `order.basket.paymentprocessorresponse_set.filter(name__in= MOBILE_PAYMENT_PROCESSORS).exists()` which I THINK might do it in 1 DB call (untested of course though)

Not a blocking comment by any means, but think it's worth thinking about potentially